src/outbox/outboxActions.js

```diff
@@ -0,0 +1,22 @@
+/* @flow */
+import type { Dispatch, GetState } from '../types';
+import { MESSAGE_SEND } from '../actionConstants';
+import { getAuth } from '../selectors';
+import { attemptToSendMessages } from './outboxMessageActions';
+
+export const sendMessageAction = (params: Object) => ({
+  type: MESSAGE_SEND,
+  params,
+});
+
+export const addToOutbox = (
+  type: 'private' | 'stream',
+  to: string | string[],
+  subject: string,
+  content: string,
+) => async (dispatch: Dispatch, getState: GetState) => {
+  dispatch(sendMessageAction({ type, to, subject, content, localMessageId: new Date().getTime() }));
```

Comment:
What we did in the webapp is to use the last message ID we have + 0.01, since we sort messages by ID in the frontend.  But from a server perspective, it can be any float; the main thing that's important is that a given client not give multiple messages the same local_id.